iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -215,10 +217,13 @@ public List<InputSplit> getSplits(JobContext context) {
 
     long fromVersion = conf.getLong(InputFormatConfig.SNAPSHOT_ID_INTERVAL_FROM, -1);
     Scan<?, FileScanTask, CombinedScanTask> scan;
+    final ExecutorService workerPool =
+            ThreadPools.newWorkerPool("iceberg-plan-worker-pool",
+                    conf.getInt(InputFormatConfig.TABLE_PLAN_WORKER_POOL_SIZE, ThreadPools.WORKER_THREAD_POOL_SIZE));
     if (fromVersion != -1) {
-      scan = applyConfig(conf, createIncrementalAppendScan(table, conf));
+      scan = applyConfig(conf, createIncrementalAppendScan(table, conf)).planWith(workerPool);
```

Comment:
Do we need to shutdown this ``workerPool`` in the ``finally`` block

Reply:
Just double check once, that iceberg takes care of that & there is no leak here before committing

Reply:
Make sense. Let me double check again.

Reply:
I checked iceberg code, and didn't find any code snippet about `workPool `shutdown. So i think you are right, shutdown this workerPool in the finally block is a more better way. Like flink iceberg side:
https://github.com/apache/iceberg/blob/cbb853073e681b4075d7c8707610dceecbee3a82/flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/source/FlinkInputFormat.java#L92-L101

Change in this commit https://github.com/apache/hive/pull/5260/commits/ef6d6840a02c0a8e96cb959f61cca23da5878845


Thanks.

Reply:
thnx, looks good